src/main/java/fi/aalto/cs/apluscourses/presentation/MainViewModel.java

```diff
@@ -118,4 +121,17 @@ public ExercisesTreeViewModel getExercises() {
   public Options getExerciseFilterOptions() {
     return exerciseFilterOptions;
   }
+
+  /**
+   * <p>Returns the name of the Card to be shown based on whether a Course object exists
+   * and is not null.</p>
+   */
+  public String getCardToBeShown() {
+    if (Optional.ofNullable(courseViewModel.get())
+            .map(BaseViewModel::getModel)
+            .orElse(null) != null) {
```

Comment:
It should be enough to test that `courseViewModel.get() != null`. `BaseViewModel::getModel` is `@NotNull`.